src/dfastbe/bank_erosion/data_models/inputs.py

```diff
@@ -268,6 +276,234 @@ def _read_river_axis(self) -> LineString:
+            cls.Parameters("VShip", None, None, True, True, None),
+            cls.Parameters("NShip", None, None, True, True, None),
+            cls.Parameters("NWave", 5, None, True, True, None),
+            cls.Parameters("Draught", None, None, True, True, None),
+            cls.Parameters("ShipType", None, [1, 2, 3], True, None, None),
+            cls.Parameters("Slope", 20, None, None, True, "slp"),
+            cls.Parameters("Reed", 0, None, None, True, "rdd"),
+        ]
+
+    def _get_discharge_parameter_definitions(self) -> List[Parameters]:
+        """Get parameter definitions for discharge parameters.
+
+        Returns:
+            List[namedtuple]: List of parameter definitions.
+        """
+        return [
+            self.Parameters("VShip", self.velocity, None, None, None, None),
+            self.Parameters("NShip", self.number, None, None, None, None),
+            self.Parameters("NWave", self.num_waves, None, None, None, None),
+            self.Parameters("Draught", self.draught, None, None, None, None),
+            self.Parameters("ShipType", self.type, [1, 2, 3], True, None, None),
+            self.Parameters("Slope", self.slope, None, None, True, "slp"),
+            self.Parameters("Reed", self.reed, None, None, True, "rdd"),
+        ]
+
+    @staticmethod
+    def _calculate_ship_derived_parameters(
+        slope_values: List[np.ndarray], reed_values: List[np.ndarray]
+    ) -> Tuple[List[np.ndarray], List[np.ndarray]]:
+        """Calculate derived parameters from slope and reed values.
+
+        Args:
+            slope_values (List[np.ndarray]): Slope values for each bank.
+            reed_values (List[np.ndarray]): Reed values for each bank.
+
+        Returns:
+            Tuple[List[np.ndarray], List[np.ndarray]]: Calculated mu_slope and mu_reed values.
+        """
+        mu_slope = []
+        mu_reed = []
+
+        for ps, pr in zip(slope_values, reed_values):
+            # Calculate mu_slope (inverse of slope for non-zero values)
+            mus = ps.copy()
+            mus[mus > 0] = 1.0 / mus[mus > 0]
+            mu_slope.append(mus)
+
+            # Calculate mu_reed (empirical damping coefficient)
+            mu_reed.append(8.5e-4 * pr**0.8)
+
+        return mu_slope, mu_reed
+
+    def read_discharge_parameters(
+        self,
+        level_i: int,
+        num_stations_per_bank: List[int],
+    ) -> SingleLevelParameters:
+        """Read Discharge level parameters.
+
+        Read all discharge-specific input arrays for level_i.
+
+        Args:
+            level_i (int):
+                The index of the discharge level.
+            num_stations_per_bank (List[int]):
+                The number of stations per bank.
+
+        Returns:
+            SingleLevelParameters: The discharge level parameters.
+        """
+        level_i_str = f"{level_i + 1}"
+
+        # Get parameter definitions
+        param_defs = self._get_discharge_parameter_definitions()
```

Comment:
there is a duplication here  with the method `get_ship_date`? can you abstract the duplication into another method and call it in both methods?